src/main/java/com/example/chapter1/basicprogramming/BinarySearchRecursively.java

```diff
@@ -0,0 +1,25 @@
+package com.example.chapter1.basicprogramming;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.Arrays;
+
+public class BinarySearchRecursively {
+
+
+    public static int rank(int[] input, int searchElement) {
+        if (ArrayUtils.isEmpty(input)) throw new IllegalArgumentException("Empty array.");
+        Arrays.sort(input);
+        return rank(input, searchElement, 0, input.length - 1);
+    }
+
+    private static int rank(int[] input, int searchElement, int low, int high) {
+        if (ArrayUtils.isEmpty(input)) throw new IllegalArgumentException("Empty array.");
+        Arrays.sort(input);
+        if (low > high) return -1;
```

Comment:
Is there any better way of representing the value which we are searching inside an array is not present instead of returning -1?
Method returns int value of search element should we return boolean ?
Can we use enums or just thrown an exception?